### PR TITLE
[REF] *: remove web.assets_common 

### PR DESCRIPTION
**Pull request by sdegueldre**
*: auth_password_policy, bus, event, im_livechat, mail, mass_mailing,
mrp_subcontracting, point_of_sale, pos_self_order, project, stock,
survey, web, web_editor, web_tour, website, website_event,
website_forum, website_sale, website_slides, base

Historically, the web.assets_common bundle was used to contain assets
that were needed by both the frontend and the backend. In practice, this
caused a bunch of issues where people would add things in assets common
that were not needed by both, and it was also abused as a way to get
bootstrap working in unrelated places by only using that bundle's css.

Because of this, as a first step, the assets_common stop being used in
the frontend, but was left everywhere else.

This commit removes the bundle completely, and moves the files that used
to be in that bundle in the other bundles that need them, this will
allow those bundles to evolve independently going forward.

in im_livechat and mail, some of the unneeded legacy code was removed, this
allows us to avoind including all of the legacy code from web in the
livechat embed bundle and in the dicuss public bundle respectively.

Enterprise: https://github.com/odoo/enterprise/pull/45884

### CODE REVIEW

File: addons/mrp_subcontracting/__manifest__.py [hunk 1]
```diff
@@ -43,12 +43,105 @@
...
+            'web/static/src/legacy/js/core/session.js',
+            'web/static/src/legacy/js/core/session_storage.js',
+            'web/static/src/legacy/js/core/time.js',
+            'web/static/src/legacy/js/core/translation.js',
+            'web/static/src/legacy/js/core/utils.js',
+            'web/static/src/legacy/js/core/widget.js',
+            'web/static/src/legacy/js/services/config.js',
+            'web/static/src/legacy/js/services/core.js',
+            'web/static/src/legacy/js/common_env.js',
+            'web/static/src/legacy/js/widgets/name_and_signature.js',
```
Comment (aab-odoo) [inline addons/mrp_subcontracting/__manifest__.py:135]: legacy/**/*.js ?

File: addons/web/__manifest__.py [hunk 2]
```diff
@@ -36,18 +36,14 @@
...
-        # TODO in the future, probably remove "assets_common" definition
-        # entirely and let all "main" bundles evolve on their own, including the
-        # files they need in their bundle.
-        'web.assets_common': [
+        'web.assets_emoji': [
+            'web/static/src/core/emoji_picker/emoji_data.js'
+        ],
+        'web.assets_backend': [
             ('include', 'web._assets_helpers'),
+            ('include', 'web._assets_backend_helpers'), # from assets backend
```
Comment (aab-odoo) [inline addons/web/__manifest__.py:46]: include assets_core here maybe?

**Review by aab-odoo — APPROVED** [review]
wow :-)

**Review by Julien00859 — APPROVED** [review]
for base

File: addons/mrp_subcontracting/__manifest__.py (hunk 1 above)
Comment (aab-odoo) [reply addons/mrp_subcontracting/__manifest__.py:132]: Fine by me :+1:

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo r+ p0

**Comment by aab-odoo** [comment]
robodoo retry

**Pull request closed without merging**